File: AnbayashiRoulette/app/src/main/java/com/example/anbayashiroulette/MainActivity.java
package com.example.anbayashiroulette;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
//import androidx.appcomat.app.ActionBarActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.cardList);

        recyclerView.setHasFixedSize(true);

        LinearLayoutManager llManager = new LinearLayoutManager(this);
        // 縦スクロール
        llManager.setOrientation(LinearLayoutManager.VERTICAL);
        recyclerView.setLayoutManager(llManager);

        ArrayList<AnbayashiData> anbayashi = new ArrayList<AnbayashiData>();
        for (int i = 0; i < MyData.commentArray.length; i++){
            anbayashi.add(new AnbayashiData(
                    MyData.numberArray[i],
                    MyData.additionArray[i],
                    MyData.commentArray[i]
            ));
        }

        RecyclerView.Adapter adapter = new AnbayashiAdapter(anbayashi);
        recyclerView.setAdapter(adapter);
        recyclerView.smoothScrollToPosition(anbayashi.size() -1); // 最後までスクロール
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu){
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item){
        // Handle action bar item clicks here, The action bar will
        // automatically handle clicks on the home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml
        int id = item.getItemId();
        if(id == R.id.action_setting){
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: SendMessage/app/src/main/java/com/example/sendmessage/MainActivity.kt
package com.example.sendmessage

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.content.Intent
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // ----- 以下を追加する -----
        // Buttonへの参照を取得する
        val button = findViewById<Button>(R.id.button)
        // ボタンタップ時の処理を指定する
        button.setOnClickListener {
            // EditTextへの参照を取得する
            val editText = findViewById<EditText>(R.id.editText)
            // メッセージを取り出す
            val message = editText.text.toString()

            // Intentを作成する
            val intent = Intent(this, ResultActivity::class.java)
            // パラメータをセットする
            intent.putExtra("message", message)
            // 画面を遷移させる
            startActivity(intent)
        }
        // ------------------------
    }
}
